indico/queries/questionnaire.py

```diff
@@ -0,0 +1,307 @@
+
+
+class _CreateQuestionaire(GraphQLRequest):
+    """
+    Creates the questionnaire (teach task) for a dataset.
+
+    Args:
+        name (str): The name of the questionnaire.
+        dataset_id (int): The id of the dataset to create the questionnaire from.
+        source_column_id (int): The id of the source column to create a questionnaire from.
+        targets (List[str]): The classes for the dataset.
+        task_type (str): The type of the task to create.
+        data_type (str): The type of the source data.
+
+    Returns:
+        Questionnaire
+
+    """
+
+    query = """
+        mutation(
+            $name: String!,
+            $dataset_id: Int!,
+            $questions: [QuestionInput]!,
+            $source_col_id: Int!,
+            $data_type: DataType!,
+        ) {
+            createQuestionnaire (
+                datasetId: $dataset_id,
+                dataType: $data_type,
+                name: $name,
+                numLabelersRequired: 1,
+                questions: $questions,
+                sourceColumnId: $source_col_id,
+                instructions: ""
+            ) {
+                id
+                questionsStatus
+                numTotalExamples
+            }
+        }
+    """
+
+    def __init__(
+        self,
+        name: str,
+        dataset_id: int,
+        source_column_id: int,
+        targets: List[str],
+        task_type: str,
+        data_type: str,
+    ):
+        questions = [
+            {"type": task_type, "targets": targets, "keywords": [], "text": name,}
+        ]
+        super().__init__(
+            query=self.query,
+            variables={
+                "name": name,
+                "dataset_id": dataset_id,
+                "questions": questions,
+                "source_col_id": source_column_id,
+                "data_type": data_type,
+            },
+        )
+
+    def process_response(self, response):
+        try:
+            questionnaire = Questionnaire(
+                **super().process_response(response)["createQuestionnaire"]
+            )
+        except IndexError:
+            raise IndicoNotFound("Failed to create Questionnaire")
+        return questionnaire
+
+
+class CreateQuestionaire(RequestChain):
+    """
+    Creates a labeled questionaire (teach task) for a dataset.
+
+    Args:
+        name (str): The name of the questionnaire.
+        targets (set): A set containing the classes/label options for the task.
+        dataset (Dataset): Dataset to create the questionnaire from.
+        target_lookup (dict): An optional dict mapping sample text to labels.
+        task_type (str): The type of the task. Defaults to ANNOTATION.
+        data_type (str): The type of the data. Defaults to TEXT.
+        
+    Returns:
+        Questionnaire object   
+    """
+
+    previous = None
+
+    def __init__(
+        self,
+        name: str,
+        targets: set,
+        dataset: Dataset,
```

Comment:
Can we use a union type of Dataset and Id here so we don't force the use of a dataset object and people can simply pass the id.